ArduPlane/navigation.cpp

```diff
@@ -252,6 +252,78 @@ void Plane::calc_gndspeed_undershoot()
     }
 }
 
+//Continuosly update / calculate effective (Ground) speed towards next WP, with ETA estimates included
+//Also keeps updating Wind vector projection split to on-bearing, prependicular to bearing, with Wind compensation angle
+//Note that these estimates may be very inaccurate in the first 3-5 seconds of turning towards the new WP
+//But they are becoming more & more accurate after 10-30 secs, as wind & airspeed fluctuations are averaging out
+//They are ment to be used as general helper variables for other, more complex navigation info / decision-making functions (for Plane)
+//Description of the continuosly updated variables are defined in Plane.h
+void Plane::Estimate_Bearing_SPD_ETA()
+{
+	  AP_Mission::Mission_Command next_cmd,c_cmd;
+	  uint16_t c_index = plane.mission.get_current_nav_index(); 
+	  uint16_t next_index=c_index+1;
+	  plane.mission.get_next_nav_cmd(c_index, c_cmd); 
+	  plane.mission.get_next_nav_cmd(next_index, next_cmd);
```

Comment:
Move variable declaration as close as possible to the first initialisation of them.